### PR TITLE
Redirect tag

### PR DESCRIPTION
**Pull request by observeralone**
add redirect tag for shadow service

### CODE REVIEW

File: plugin-api/pom.xml [hunk 1]
```diff
@@ -91,6 +97,27 @@
                     </execution>
                 </executions>
             </plugin>
+
+            <plugin>
```
Comment (akwei) [inline plugin-api/pom.xml:101]: We can use a class to get data from System env. For example:
```java

public class EnvInfo {

	private Map<String,String> map;

	ignore getter setter


	public String getInfo(String key){
		return this.map.get(key);
	}

}

```
This class can be isolated from getting environment variables.

In unit test:

```java

EnvInfo envInfo = new EnvInfo();

// mock data in system environment
envInfo.setMap();

```

Other classes can get data from EnvInfo.

Comment (observeralone) [reply plugin-api/pom.xml:101]: good idea, i'll change it

**Review by Oseenix — COMMENTED** [review]
There may increase many logs when the redirect is applied.  Maybe these logs can be deleted or applied a rate limit logger.

**Comment by observeralone** [comment]
> There may increase many logs when the redirect is applied. Maybe these logs can be deleted or applied a rate limit logger.

The redirected log output will only be output during initialization, and its output log will not be too much

**Review by Oseenix — APPROVED** [review]

**Review by akwei — APPROVED** [review]

**Comment by observeralone** [comment]
close #184

**Pull request merged**